lib/fulfil/rate_limit.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module Fulfil
+  # The `Fulfil::RateLimit` allows clients to keep track of their API usage.
+  class RateLimit
+    attr_reader :limit, :requests_left, :resets_at
+
+    def initialize
+      @limit = 0
+      @requests_left = 0
+      @resets_at = nil
+    end
+
+    # Analyses the rate limit based on the response headers from Fulfil.
```

Comment:
Love the documentation. I'll have to get in to this habit 😄

Reply:
For gems, I actually like adding documentation. Gems are often a bit more abstract than application code. So, I figured it made sense to add it here too.

Reply:
It definitely helps someone new to the code get a better footing.